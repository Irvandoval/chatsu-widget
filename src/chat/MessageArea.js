import React, { Component } from 'react';
import dateFormat from 'dateformat';
import "./chat.css"

const dayInMillis = 60 * 60 * 24 * 1000;

export default class MessageArea extends Component {
  componentDidMount() {
    window.scrollTo(0, document.body.scrollHeight);
  }

  componentDidUpdate() {
      window.scrollTo(0, document.body.scrollHeight);
  }

  render() {
    const currentTime = new Date();

    return (
      <ol class="chat">
        {this.props.messages.map(({name, text, from, time}) => {
          if (from === 'visitor') {
            name = "You"
          }
          return (
            <li class={from}>
              <div class="msg">
                <p>{name ? name + ": " + text : text}</p>
                { (this.props.conf.displayMessageTime) ?
                    <div class="time">
                      {
                        currentTime - new Date(time) < dayInMillis ?
                          dateFormat(time, "HH:MM") :
                          dateFormat(time, "m/d/yy HH:MM")
                      }
                    </div>
                    :
                    ''
                }
              </div>
            </li>
          );
        })}
      </ol>
    );
  }
}