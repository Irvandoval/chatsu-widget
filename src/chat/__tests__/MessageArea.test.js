import React from 'react';
import MessageArea from '../MessageArea';
import TestRenderer from 'react-test-renderer';
import { JSDOM } from 'jsdom';

const dom = new JSDOM();
global.document = dom.window.document;
global.window = dom.window;
global.window.scrollTo = function() {
    return;
};

describe('<MessageArea />', () => {
    it('matches the snapshot', () => {
      const messages = [];
      const tree = TestRenderer.create(<MessageArea messages={messages}/>);
      expect(tree).toMatchSnapshot();
    });
})