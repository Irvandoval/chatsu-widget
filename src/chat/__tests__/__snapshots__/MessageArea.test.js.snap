// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageArea /> matches the snapshot 1`] = `
<ol
  class="chat"
/>
`;
